### PR TITLE
Define `Question` interface and generate schema

### PR DESCRIPTION
**Pull request by chrichri17**
This solves #19

### CODE REVIEW

**Comment by chrichri17** [comment]
Current dependencies on/for this PR:
* main
  * **PR #27** <a href="https://app.graphite.dev/github/pr/YacineSteeve/quizoot/27" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/YacineSteeve/quizoot/27?utm_source=stack-comment).

File: schemas/interface.ts [hunk 1]
```diff
@@ -1,18 +1,186 @@
...
-     * @pattern [a-z0-9]{12}
+     * @version 0.1.0
+     */
+    export interface Question {
+        /**
+         * Id of the question
+         * 
+         * @pattern [a-z0-9]{12}
+         */
+        id: string;
```
Comment (YacineSteeve) [inline schemas/interface.ts:13]: Why do you use strings instead of numbers for your `id` props ?

File: schemas/question.json [hunk 2]
```diff
@@ -1,17 +1,243 @@
...
+                    "$ref": "#/definitions/Quizoot.MultipleChoicesQuestion"
+                },
+                {
+                    "$ref": "#/definitions/Quizoot.TextQuestion"
+                },
+                {
+                    "$ref": "#/definitions/Quizoot.UploadQuestion"
+                },
+                {
+                    "properties": {
```
Comment (YacineSteeve) [inline schemas/question.json]: Seems not uniform...

File: schemas/interface.ts [hunk 3]
```diff
@@ -1,18 +1,186 @@
...
+    }
+
     /**
-     * The actual question as a human readable string.
+     * Enum of supported file extensions.
      */
-    question: string;
+    type FileType = ".txt" | ".csv" | ".pdf" | ".img" | ".jpg" | ".jpeg" | ".png";
+
+    type CodeQuestion = {
```
Comment (YacineSteeve) [inline schemas/interface.ts]: You defined `ChoiceQuestion`, `UploadQuestion` and `TextQuestion` as interfaces, but `CodeQuestion` as a type ? 
Probably leading to [this](https://github.com/YacineSteeve/quizoot/pull/27/files#r1029766789)

File: schemas/interface.ts [hunk 4]
```diff
@@ -1,18 +1,186 @@
...
+        /**
+         * Code content. This will be ran using a `python3` runner and evaluated.
+         */
+        content: string;
+        /**
+         * Coding language. Only `python3` is supported.
+         */
+        language: "python3";
```
Comment (YacineSteeve) [inline schemas/interface.ts:180]: Did you add `language` as a property for 'reusability' reasons ? 😇 
As we know that only `python3` will be supported here...

**Review by YacineSteeve — APPROVED** [review]
Looks good.

File: schemas/interface.ts (hunk 1 above)
Comment (chrichri17) [reply schemas/interface.ts:13]: Good point. We have more ids if we use strings haha. I don't know how many questions we might have so.. just to be extremely safe

File: schemas/interface.ts (hunk 3 above)
Comment (chrichri17) [reply schemas/interface.ts]: Oh my bad. Ty. I'll change that

File: schemas/interface.ts (hunk 4 above)
Comment (chrichri17) [reply schemas/interface.ts:180]: We might want to add more, indeed. Like `R`

Comment (YacineSteeve) [reply schemas/interface.ts:180]: yeah, I see

File: schemas/interface.ts (hunk 1 above)
Comment (YacineSteeve) [reply schemas/interface.ts:13]: A database of billions of questions 🤣 why not

**Pull request merged**